src/meta/model/migration/src/m20251124_195858_pending_sink_state.rs

```diff
@@ -0,0 +1,156 @@
+use sea_orm_migration::prelude::*;
+
+use crate::drop_tables;
+use crate::sea_orm::{FromQueryResult, Statement};
+use crate::utils::ColumnDefExt;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[derive(FromQueryResult, Debug)]
+struct OldRow {
+    pub sink_id: i32,
+    pub end_epoch: i64,
+    pub metadata: Vec<u8>,
+    pub snapshot_id: i64,
+    pub committed: bool,
+}
+
+// Logic referenced from src/connector/src/sink/iceberg/mod.rs
+fn transform_metadata(metadata: Vec<u8>, snapshot_id: i64) -> Vec<u8> {
+    let mut write_results_bytes: Vec<Vec<u8>> = serde_json::from_slice(&metadata).unwrap();
+    let snapshot_id_bytes: Vec<u8> = snapshot_id.to_le_bytes().to_vec();
+    write_results_bytes.push(snapshot_id_bytes);
+    serde_json::to_vec(&write_results_bytes).unwrap()
+}
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(PendingSinkState::Table)
+                    .if_not_exists()
+                    .col(
+                        ColumnDef::new(PendingSinkState::SinkId)
+                            .integer()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(PendingSinkState::Epoch)
+                            .big_integer()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(PendingSinkState::SinkState)
+                            .string()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(PendingSinkState::Metadata)
+                            .rw_binary(manager)
+                            .not_null(),
+                    )
+                    .primary_key(
+                        Index::create()
+                            .col(PendingSinkState::SinkId)
+                            .col(PendingSinkState::Epoch),
+                    )
+                    .to_owned(),
+            )
+            .await?;
+
+        if manager
+            .has_table(ExactlyOnceIcebergSinkMetadata::Table.to_string())
```

Comment:
@wcy-fdu Can you review the migration logic?

Besides, have we tested the backward compatibility of this migration either locally or in the CI backward test?